### PR TITLE
Updates to Component API docs

### PR DESCRIPTION
**Pull request by psavkar**

### CODE REVIEW

File: COMPONENT-API-OLD.md [hunk 1]
```diff
@@ -0,0 +1,376 @@
+# Component API
```
Comment (dylburger) [inline COMPONENT-API-OLD.md:1]: I'd just recommend removing this file entirely to reduce any confusion. We can always recover content from git.

File: COMPONENT-API.md [hunk 2]
```diff
@@ -1,376 +1,864 @@
...
-  - [Retrieving events programmatically](#retrieving-events-programmatically)
-- [`$.service.db`](#servicedb)
-  - [set method](#set-method)
-  - [get method](#get-method)
-- [Using npm packages](#using-npm-packages)
-- [How components run](#how-components-run)
+## Diagram
 
-<!--te-->
+![image-20200816223906765](./images/image-20200816223906765.png)
```
Comment (dylburger) [inline COMPONENT-API.md:122]: When I view this file in the Github UI, this image didn't render - is that expected?

File: COMPONENT-API.md [hunk 3]
```diff
@@ -1,376 +1,864 @@
...
 
-## `name`
+## States
 
-The name of the component, a **string** which identifies components deployed to users' accounts.
+Pipedream components support `Activate()` and `Deactivate()` lifecycle hooks. The code for these hooks are defined within the component. Learn more about the [component structure](#component-structure) and [hook usage](#hooks).
```
Comment (dylburger) [inline COMPONENT-API.md:126]: lowercase `activate` and `deactivate`

File: COMPONENT-API-OLD.md [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+};
+```
+
+The component API borrows concepts and structure from frontend components, like those in [Vue.js](https://vuejs.org/) or [React](https://reactjs.org/).
+
+Components [run on Pipedream infrastructure](#how-components-run).
+
+## Features
+
+Component can:
```
Comment (dylburger) [inline COMPONENT-API-OLD.md:25]: Component -> Components

File: COMPONENT-API.md [hunk 5]
```diff
@@ -1,376 +1,864 @@
...
 
-This name will show up in the Pipedream UI, in CLI output (for example, from `pd list` commands), etc.
+### Saved Component
 
-### Name slugs
+A saved component is non-instantiated component code that has previously been deployed to Pipedream. Each saved component has a unique saved component ID. Saved components cannot be invoked directly — they must first be depoyed.
```
Comment (dylburger) [inline COMPONENT-API.md:130]: "they must first be depoyed" -> deployed

File: COMPONENT-API.md [hunk 6]
```diff
@@ -1,376 +1,864 @@
...
+
+The `Deactivate()` hook is automatically invoked by Pipedream when a component is updated or deleted. A common use case for the deactivate hook is to automatically delete a webhook subscription when a component is deleted, but you can run any valid code. To learn more about defining a custom `Deactivate()` hook, refer to the [API documentation](#hooks-1). 
+
+## Management
+
+### CLI
+
+#### Development Mode
+
+The easiest way to develop and iteratively test is to use the `pd dev` command tol deploy a local file, attach to a component, and automatically update the component on each local save. To deploy a new component with `pd dev`, run:
```
Comment (dylburger) [inline COMPONENT-API.md:172]: "tol deploy" -> to

File: COMPONENT-API.md [hunk 7]
```diff
@@ -1,376 +1,864 @@
...
 ```bash
-pd events <deployed-component-name>
+pd deploy my-component.js
 ```
 
-## Referencing `this`
+##### From Pipedream Github Repo
 
-`this` refers to the component, and provides access to [props](#props), [methods](#methods), and Pipedream-provided functions like [`this.$emit`](#thisemit).
+You can explore the components available to deploy in [Pipedream's Github repo](https://github.com/PipedreamHQ/pipedream/tree/master/components).
```
Comment (dylburger) [inline COMPONENT-API.md:202]: Maybe make this URL relative?

File: COMPONENT-API.md [hunk 8]
```diff
@@ -1,376 +1,864 @@
...
+
+#### Delete
+
+View the [CLI command reference](https://docs.pipedream.com/cli/reference/#command-reference).
+
+### UI
+
+#### Deploy
+
+You can find any deploy curated components at https://pipedream.com/sources/new, or you can deploy code via the UI using following URL patterns.
```
Comment (dylburger) [inline COMPONENT-API.md:238]: "You can find any deploy curated components" -> did you mean "find and deploy"?

File: COMPONENT-API.md [hunk 9]
```diff
@@ -1,376 +1,864 @@
...
+
+## Diagram
+
+![image-20200811161858648](images/image-20200811161858648.png)
+
+## Triggering Components
+
+Components are triggered when you manually run them (e.g., via the **RUN NOW** button in the UI) or when one of their [interfaces](#interface-props) is triggered. Pipedream currently support **HTTP** and **timer** interfaces.
+
+When a component is triggered, the `run()` method of the component is executed. Standard output and errors are surfaced in the **logs**.
```
Comment (dylburger) [inline COMPONENT-API.md:289]: Logs tab?

File: COMPONENT-API.md (hunk 2 above)
Comment (psavkar) [reply COMPONENT-API.md:122]: no -- it worked in my personal repo before i merged. i will fix. thanks!

File: COMPONENT-API.md (hunk 3 above)
Comment (psavkar) [reply COMPONENT-API.md:126]: ack

File: COMPONENT-API.md (hunk 5 above)
Comment (psavkar) [reply COMPONENT-API.md:130]: ack

File: COMPONENT-API.md (hunk 6 above)
Comment (psavkar) [reply COMPONENT-API.md:172]: ack

File: COMPONENT-API.md (hunk 7 above)
Comment (psavkar) [reply COMPONENT-API.md:202]: ack

File: COMPONENT-API.md (hunk 8 above)
Comment (psavkar) [reply COMPONENT-API.md:238]: yes - thx!

File: COMPONENT-API.md (hunk 9 above)
Comment (psavkar) [reply COMPONENT-API.md:289]: ack

**Pull request merged**